Objects/unicodeobject.c

```diff
@@ -14270,6 +14270,147 @@ unicode_getnewargs(PyObject *v, PyObject *Py_UNUSED(ignored))
     return Py_BuildValue("(N)", copy);
 }
 
+/* Dedent a string.
+   Behaviour is expected to be an exact match of `textwrap.dedent`.
+   Return a new reference on success, NULL with exception set on error.
+   */
+PyAPI_FUNC(PyObject *)
+_PyUnicode_Dedent(PyObject *unicode)
+{
+    Py_ssize_t src_len = 0;
+    const char *src = PyUnicode_AsUTF8AndSize(unicode, &src_len);
+    if (!src) {
+        return NULL;
+    }
+    if (src_len <= 0) {
```

Comment:
When can `src_len` be negative?

Reply:
when PyUnicode_AsUTF8AndSize got some internal errors 👀

Reply:
I will change it to an assert